src/components/TokenIcon.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { useSelector } from 'react-redux'
+import { AbstractPosition } from 'src/positions/types'
+import { tokensListSelector } from 'src/tokens/selectors'
+import { BaseToken } from 'src/tokens/slice'
+
+interface Props {
+  token: BaseToken | AbstractPosition
+  viewStyle?: StyleProp<ViewStyle>
+  testID?: string
+}
+
+// Temporary until isNative is populated throughout address metadata
+const nativeTokenAddresses = [
+  '0x471ece3750da237f93b8e339c536989b8978a438', // Celo Mainnet
+  '0xf194afdf50b03e69bd7d057c1aa9e10c9954e4c9', // Celo Alfajores
+]
+
+export default function TokenIcon({ token, viewStyle, testID }: Props) {
+  const isToken = 'symbol' in token
+  const tokens = useSelector(tokensListSelector)
+
+  // Always defaults to Celo as token.network is not currently defined
+  // Switch statement for easier future expansion
+  let networkImage
+  switch (token.network) {
+    default:
+    case 'Celo':
```

Comment:
would be nice to avoid the switch here with hardcoded network token names.. Maybe this logic can go in the cloud function / blockchain api route to get tokens and the wallet directly can access an optional "subscript logo url" (could use a better name) which can be used here? That allows us flexibility on switching this logic too in the future without a new version release.

Reply:
I agree that it would be nice to this hardcoded switch. I do worry storing another key value for every entry might be get quite large when we already have the network to use, but if that's unfounded it certainly seems easiest.

Maybe this is a good case for a small dynamic config with some basic network information dynamically: this would be network and not chain specific. Also because its just general information it would not need to be app version specific as was the concern in https://github.com/valora-inc/wallet/pull/4115#discussion_r1309727228.

```JSON
{
  "celo": {
    "nativeSymbol": "CELO",
    "iconUrl": "https://example.com/celo-icon.png",
    "foundationWebsite": "https://example.com",
  },
  "ethereum": {
    "nativeSymbol": "ETH",
    "iconUrl": "https://example.org/icon.png",
    "foundationWebsite": "https://example.org",
  }
}
```